qemu/tests/qemu_guest_agent.py

```diff
@@ -1627,6 +1627,64 @@ def check_vss_info(cmd_type, key, expect_value):
                 test.error("guest-fsfreeze-thaw cmd failed with:"
                            "('%s')" % str(detail))
 
+    @error_context.context_aware
+    def gagent_check_fsinfo(self, test, params, env):
+        """
+        Execute "guest-fsinfo" command to guest agent,check file system of
+        mountpoint,disk's name and serial number.
+
+        steps:
+        1) check file system type of every mount point.
+        2) check disk name.
+        3) check disk's serial number.
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+
+        """
+        session = self._get_session(params, None)
+        self._open_session_list.append(session)
+
+        error_context.context("Check all file system info in a loop.", logging.info)
+        fs_info_qga = self.gagent.get_fsinfo()
+        for fs in fs_info_qga:
+            mount_pt = fs["mountpoint"]
+            if params["os_type"] == "windows":
+                mount_pt = mount_pt[:2]
+
+            error_context.context("Check file system type of '%s' mount point." %
+                                  mount_pt, logging.info)
+            fs_type_qga = fs["type"]
+            cmd_get_disk = params["cmd_get_disk"] % mount_pt
+            disk_info_guest = session.cmd(cmd_get_disk).strip().split()
+            fs_type_guest = disk_info_guest[1]
+            if fs_type_qga != fs_type_guest:
+                test.fail("File System doesn't match.\n"
+                          "from guest-agent is %s,from guest os is %s."
+                          % (fs_type_qga, fs_type_guest))
+
+            error_context.context("Check disk name.", logging.info)
+            disk_name_qga = fs["name"]
+            disk_name_guest = disk_info_guest[0]
+            if params["os_type"] == "linux":
+                if not re.findall(r'^/\w*/\w*$', disk_name_guest):
+                    disk_name_guest = session.cmd("readlink %s" % disk_name_guest).strip()
+                disk_name_guest = disk_name_guest.split('/')[-1]
+            if disk_name_qga != disk_name_guest:
+                test.fail("Device name doesn't match.\n"
+                          "name from guest-agent is %s,name from guest os is %s."
+                          % (disk_name_qga, disk_name_guest))
+
+            error_context.context("Check serial number of some disk.", logging.info)
+            fs_check_list = ["C:\\", "a:\\", "/", "/boot"]
```

Comment:
what is `a:\\` ? My test result is `C:\\` and `S:\\` have `serial: GAGENT_TEST` for windows guest,
```
[{'name': '\\\\?\\Volume{b5533b98-a713-11e9-a82a-806e6f6e6963}\\', 'total-bytes': 680312832, 'mountpoint': 'E:\\', 'disk': [{'bus-type': 'sas', 'bus': 0, 'unit': 0, 'pci-controller': {'bus': -1, 'slot': -1, 'domain': -1, 'function': -1}, 'dev': '\\\\?\\Volume{b5533b98-a713-11e9-a82a-806e6f6e6963}', 'target': 2}], 'used-bytes': 680312832, 'type': 'CDFS'},
{'name': '\\\\?\\Volume{b5533b97-a713-11e9-a82a-806e6f6e6963}\\', 'total-bytes': 2936563712, 'mountpoint': 'D:\\', 'disk': [{'bus-type': 'sas', 'bus': 0, 'unit': 0, 'pci-controller': {'bus': -1, 'slot': -1, 'domain': -1, 'function': -1}, 'dev': '\\\\?\\Volume{b5533b97-a713-11e9-a82a-806e6f6e6963}', 'target': 1}], 'used-bytes': 2936563712, 'type': 'CDFS'},
{'name': '\\\\?\\Volume{d8d25900-0000-0000-0000-d01200000000}\\', 'total-bytes': 31895580672, 'mountpoint': 'C:\\', 'disk': [{'serial': 'GAGENT_TEST', 'bus-type': 'sas', 'bus': 0, 'unit': 0, 'pci-controller': {'bus': -1, 'slot': -1, 'domain': -1, 'function': -1}, 'dev': '\\\\.\\PhysicalDrive0', 'target': 0}], 'used-bytes': 15079198720, 'type': 'NTFS'},
{'name': '\\\\?\\Volume{d8d25900-0000-0000-0000-100000000000}\\', 'total-bytes': 314568704, 'mountpoint': 'S:\\', 'disk': [{'serial': 'GAGENT_TEST', 'bus-type': 'sas', 'bus': 0, 'unit': 0, 'pci-controller': {'bus': -1, 'slot': -1, 'domain': -1, 'function': -1}, 'dev': '\\\\.\\PhysicalDrive0', 'target': 0}], 'used-bytes': 35090432, 'type': 'NTFS'}]
```

Reply:
It's a typo, it should be S:\\.